### ISSUE TITLE
Split implementations for Gevent and Asyncio and deprecate Gevent

### ISSUE THREAD

**Issue by kvakvs** [esl/Pyrlang#23]
- [x] Migrate logging to standard Python logger
- [x] Offer implementation of backend for Gevent
- [x] Offer implementation of backend for Asyncio

**Comment by s2hc-johan** [comment]
I like the ide of making clear abstraction layer so we can switch the underlying aync lib. Bbut why drop gevent at this point and pushing for python 3.5+ only?

**Comment by kvakvs** [comment]
Now it is two branches, I can run changes in them in parallel, but if there is no demand for Python2 and gevent then there is no point in keeping it alive?

**Comment by s2hc-johan** [comment]
Agreed, didn't know you had started looking at asyncio, thought we skipped it since the API is changing between releases still. But we shouldn't do twice the amount of work needed!

**Comment by vegabook** [comment]
wouldn't it be better simply to use python threads? They seem to me to map better to the idea of Erlang processes than async techniques like asyncio or gevent.

**Comment by s2hc-johan** [comment]
The reason why I started to implement this was when I did work for a client of mine. I wrote some worker code that had to fold over big amount of data and I implemented it with threads, or it might have been processes in pyhon2. When doing this the performance were quite bad and the were a lot of nasty code to keep queues in check and seeing if any process died. That and you loose site of the state in python quite fast unless your debug fu is good. I then built the cordinating code in erlang and called out to 2/4/8 python processes depending on how much calculation power I wanted. This was so much cleaner and easier to reason about and develop with that I started to look at a more generic solution.

I'm not that in to python3 and how much better threds/processes is working now. But I think it's dangerous to take the concept of erlang process to something that is much more heavy. We already have the impact  of faking erlang flow ontop of python.

Also my initial ideas for this there is nothing stopping you from starting 10 python processes and let them be 1 node each. By keeping it simple in python land only accessing it when you want to do the stuff it's good at, you have the benefit of erlang shell, monitoring etc. Yes we can implement this in python as well, but the idea is to use erlang where erlang shines and python where python shines. Rather then implementing Erlang design and monitoring features on top of python threads

An example, in python 2 again. Gunicorn, celery start a main process that then starts worker threads/processes. Celery have a lot of code to be able to control the state of the workers, numbers etc. All those things you get for free in erlang shell. Gunicorn could be implemented in erlang giving huge possibility for live debugging of flow, while we still have the standard wsgi application in python running synchronous.
